### PR TITLE
Add atan2

### PR DESCRIPTION
**Pull request by mattfbacon**
Maybe one point of contention is whether it should be (x, y) or (y, x).

### CODE REVIEW

**Comment by CGMossa** [comment]
In my opinion atan2 should always be y then x. It is like that everywhere I've looked.

This is needed for the tikz esque endeavor people are on.

**Comment by Enivex** [comment]
> In my opinion atan2 should always be y then x. It is like that everywhere I've looked.
> 
> This is needed for the tikz esque endeavor people are on.

(x,y) makes way more sense. What you're doing is finding the argument of x+iy.
 
https://reference.wolfram.com/language/ref/ArcTan.html

**Comment by Enivex** [comment]
> I believe they were talking about atan2, not the order of arguments

They're talking about whether the angle between (x,y) and the x-axis should be computed with atan2(x,y) or atan2(y,x)

**Comment by laurmaedje** [comment]
Happy to merge this once there's consensus on the argument order.

**Comment by CGMossa** [comment]
Does atan and atan2 return the same thing and have the same range? I think not?

**Comment by Enivex** [comment]
> Does atan and atan2 return the same thing and have the same range? I think not? 

Both return angles, but the range of atan2 is (-pi,pi], while atan is specifically the inverse of tan on (-pi/2,pi/2).

In fact atan2(1,x) = atan(x) (or the reverse order)

The main reason why some use atan2(y,x) is for historical reasons, as that's what fortran used.

**Comment by CGMossa** [comment]
I revoke my vote on `atan2(y,x)` and now fully support `atan2(x,y)`, but with a historic note in the docs.
typst is already cleaning up a lot of "historic reasons" garbage. Why stop.

I got swayed by @Enivex.

**Comment by mattfbacon** [comment]
Ok, if we're agreed on `x, y` I think we're all set. @laurmaedje

File: library/src/compute/calc.rs [hunk 1]
```diff
@@ -270,6 +271,27 @@ pub fn atan(
     Value::Angle(Angle::rad(value.float().atan()))
 }
 
+/// Calculate the four-quadrant arctangent of a coordinate.
+///
+/// ## Example
+/// ```example
+/// #calc.atan(1, 1) \
+/// #calc.atan(-2, -3)
```
Comment (laurmaedje) [inline library/src/compute/calc.rs:279]:
```suggestion
/// #calc.atan2(1, 1) \
/// #calc.atan2(-2, -3)
```

File: library/src/compute/calc.rs [hunk 2]
```diff
@@ -270,6 +271,27 @@ pub fn atan(
     Value::Angle(Angle::rad(value.float().atan()))
 }
 
+/// Calculate the four-quadrant arctangent of a coordinate.
+///
```
Comment (laurmaedje) [inline library/src/compute/calc.rs:275]: A note regarding argument order would be great.

File: library/src/compute/calc.rs (hunk 1 above)
Comment (mattfbacon) [reply library/src/compute/calc.rs:279]: Weird, I thought I fixed that. Must have messed up the git stuff. Anyway, fixed now.

**Pull request merged**